projects/kit/components/carousel/carousel.component.ts

```diff
@@ -54,9 +54,11 @@ import {TuiCarouselScroll} from './carousel-scroll.directive';
         '[class._transitioned]': 'transitioned',
         '[class._draggable]': 'draggable',
         '(touchstart)': 'onTransitioned(false)',
+        '(document:touchmove)': 'transitioned ? (0) : shift()',
         '(touchend)': 'onTransitioned(true)',
+        '(document:mousemove)': 'transitioned ? (0) : shift()',
         '(mousedown)': 'onTransitioned(false)',
-        '(document:mouseup.zoneless)': 'onTransitioned(true)',
+        '(document:mouseup.zoneless)': 'onTransitioned(true);',
```

Comment:
I think we do not really need those event changes, we can just emit values inside `onPan` and `onTransitioned`

Reply:
fixed